kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -347,7 +367,15 @@ private Table retrieveTable(TableId tableId) {
     try {
       return getBigQuery().getTable(tableId);
     } catch (BigQueryException e) {
-      if (BigQueryErrorResponses.isIOError(e)) {
+      /* 1. Authentication error thrown by bigquery is a type of IOException
+       and the error code is 0. That's why we create a separate
+       check function for Authentication error otherwise this falls under IOError check */
+
+      /* 2. For Authentication, we don't need Retry logic. Instead, we throw Bigquery exception directly. */
+
+      if (BigQueryErrorResponses.isAuthenticationError(e)) {
+        throw new BigQueryException(e.getCode(), "Failed to authenticate client for table " + tableId + " with error " + e);
```

Comment:
Either `BigQueryConnectException` ?
Or lets re-throw the original `e`

Reply:
I think `BigQueryConnectException` is better. That way, from the message, we can notify customer before hand, that this is mainly an Authentication issue and the actual error can follow.